JabbR/Nancy/AccountModule.cs

```
@@ -123,6 +124,24 @@ public class AccountModule : JabbRModule
                     return View["register", ModelValidationResult];
                 }
             };
+
+            Get["/unlink/{providerName}"] = param =>
+                {
+                    string provider = param["providerName"].ToString();
+                    ChatUser user = repository.GetUserById(Context.CurrentUser.UserName);
+
+                    var identity = user.Identities.ToList()
+                        .Find(ident => ident.ProviderName.Equals(provider, StringComparison.InvariantCultureIgnoreCase));
+                    
+                    if (identity != null)
+                    {
+                        repository.Remove(identity);
+
+                        return Response.AsRedirect("~/account");
+                    }
+                    ModelValidationResult.AddError("_FORM", String.Format("Unable to find identity for provider: {0}.", provider));
```

Get rid of this. If it's an unconfigured provider we should just throw an exception inline or return 400.